api/controllers/routes/agency.js

```diff
@@ -0,0 +1,34 @@
+const express = require("express");
+const app = express();
+const requestUtils = require("../../utils/request");
+const queryUtils = require("../../utils/query");
+
+const env = process.env.NODE_ENV || 'development';
+
+const config = require("../../../config")[env];
+const host = config.host;
+const agencyResourceId = config.agency_resource;
+const agencyServiceResourceId = config.agency_service_resource;
+const serviceTaxonomyResourceId = config.service_taxonomy_resource;
+
+const uri = `${host}/api/3/action/datastore_search_sql?sql=SELECT * from "${agencyResourceId}" agency `;
+
+app.get("/", function (req, res) {
+    let query = '';
+    if (req.query.taxonomyId) {
+        query =
+            `
+            INNER JOIN "${agencyServiceResourceId}" agency_service ON agency."AGENCY_ID" = agency_service."AGENCY_ID" 
+            INNER JOIN "${serviceTaxonomyResourceId}" service_taxonomy ON agency_service."AGENCY_ID" = service_taxonomy."AGENCY_ID" 
+            AND agency_service."LINE_NUMBER" = service_taxonomy."LINE_NUMBER" 
```

Comment:
This is a curious here but why are certain words "LINE_NUMBER" in quotes?

Reply:
This is just the syntax that ckan was looking for. I had to play around with the format of the variable names to get it to work, and this ended up solving it